c7n/resources/account.py

```diff
@@ -656,3 +656,42 @@ def process(self, accounts):
             if update_args:
                 update_args['Name'] = trail_name
                 client.update_trail(**update_args)
+
+
+@filters.register('has-virtual-mfa')
+class HasVirtualMFA(Filter):
+    """Is the account configured with a virtual MFA device?
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+                - name: account-with-virtual-mfa
+                  resource: account
+                  region: us-east-1
+                  filters:
+                    - type: has-virtual-mfa
+                      value: true
+    """
+
+    schema = type_schema('has-virtual-mfa', **{'value': {'type': 'boolean'}})
+
+    permissions = ('iam:ListVirtualMFADevices',)
+
+    def mfa_belongs_to_root_account(self, mfa):
+        return mfa['SerialNumber'].endswith(':mfa/root-account-mfa-device')
+
+    def account_has_virtual_mfa(self, account):
+        if not account.get('c7n:VirtualMFADevices'):
+            client = local_session(self.manager.session_factory).client('iam')
```

Comment:
wondering if we should go ahead and turn this into an iam resource (`iam-virtual-mfa`)